database2/ffldb/ldb/transaction.go

```diff
@@ -0,0 +1,94 @@
+package ldb
+
+import (
+	"errors"
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+// LevelDBTransaction is a thin wrapper around native leveldb
+// batches and snapshots. It supports both get and put.
+// Note: transactions provide data consistency over the state of
+// the database as it was when the transaction started. There is
+// NO guarantee that if one puts data into the transaction then
```

Comment:
There's actually guarantee that it won't happen, no?

Reply:
And btw, did you check that it doesn't happen in our codebase?

Reply:
> And btw, did you check that it doesn't happen in our codebase?

Yep. Never happens as far as I can tell.